File: src/main/resources/static/js/components/company/company.service.js
(function () {
    'use strict';
    angular
        .module('myApp.company')
        .factory('companyService', companyService);

    companyService.$inject = ['$http', '$log'];

    function companyService ($http, $log) {
        var URL = '/company';
        return {
            getAll: getAll,
            create: create,
            get: get,
            update: update,
            remove: remove
        };

        ////////////////
        function getAll(){
            return $http.get(URL);
        }

        function create (form) {
            return $http.post(URL, form);
        }

        function get(id){
            return $http.get(URL + '/' + id);
        }

        function update(id, form){
            return $http.post(URL + '/' + id, form);
        }

        function remove(id){
            return $http.delete(URL + '/' + id);
        }
    }
})();

File: src/main/java/com/lpc/company/service/CompanyService.java
package com.lpc.company.service;

import com.lpc.company.model.Company;
import com.lpc.company.web.CompanyForm;
import org.springframework.transaction.annotation.Transactional;

import java.util.List;
import java.util.Optional;

public interface CompanyService {
    public List<Company> getAll();

    public Company create(CompanyForm form);

    public Optional<Company> get(Integer id);

    public Optional<Company> update(Integer id, CompanyForm form);

    @Transactional
    public void delete(Integer id);
}
